onmt/modules/Ensemble.py

```diff
@@ -99,3 +112,38 @@ def forward(self, hidden):
                          for (i, model_generator)
                          in enumerate(self.model_generators)]
         return torch.stack(distributions).mean(0)
+
+
+class EnsembleModel(NMTModel):
+    """ Dummy NMTModel wrapping individual real NMTModels """
+    def __init__(self, models):
+        encoder = EnsembleEncoder(model.encoder for model in models)
+        decoder = EnsembleDecoder(model.decoder for model in models)
+        super(EnsembleModel, self).__init__(encoder, decoder)
+        self.generator = EnsembleGenerator(model.generator for model in models)
+        self.models = nn.ModuleList(models)
+
+
+def load_test_model(opt, dummy_opt):
+    """ Read in multiple models for ensemble """
+    shared_fields = None
+    shared_model_opt = None
+    models = []
+    for model_path in opt.models:
+        fields, model, model_opt = \
+            onmt.ModelConstructor.load_test_model(opt,
+                                                  dummy_opt,
+                                                  model_path=model_path)
+        if shared_fields is None:
+            shared_fields = fields
+        else:
+            for key, field in fields.items():
+                if field is not None and 'vocab' in field.__dict__:
+                    assert field.vocab.stoi == shared_fields[key].vocab.stoi, \
+                        'Ensemble models must use the same preprocessed data'
+        models.append(model)
+        if shared_model_opt is None:
+            shared_model_opt = model_opt
+        # FIXME: anything to check or copy from other model_opt?
```

Comment:
Has this #FIXME Repaired?

Reply:
At least for now, the only value of `model_opt` that is accessed after this point is `copy_attn`.
It is not possible to ensemble models using copy attention with those not using it.
A check could be added, but in that case checks should be added also for other incompatible configurations, which is too many.

I'm considering this FIXME resolved for now.